### ISSUE TITLE
Cached resolution doesn't handle test-dependency in multi-project build

### ISSUE THREAD

**Issue by fredriv** [sbt/sbt#1730]
I have a multi-project build where one subproject contains common test-code, which the other subprojects depend on in their test configuration, e.g.:

```
lazy val foo = project.in(file("foo"))

lazy val bar = project.in(file("bar")).dependsOn(foo % Test)
```

This build fails in SBT 0.13.7-RC5 if I enable cached resolution (it works fine otherwise). Basically, it does not include the compile dependencies from "foo" in the test configuration in "bar".

I've created a minimal test project that reproduces the bug at https://github.com/fredriv/sbt-cached-resolution-bug

Not sure if this is related to other cached resolution issues such as #1723 or if this is specific to multi-project builds.

**Comment by eed3si9n** [comment]
Probably same as #1711.

**Comment by fredriv** [comment]
I ran the test using SBT 0.13.7-RC5 which was released on Nov. 12th. https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/sbt-launch/0.13.7-RC5/

It seems the fixes for #1711 were merged on Nov. 10th?

**Comment by eed3si9n** [comment]
> I've created a minimal test project that reproduces the bug at https://github.com/fredriv/sbt-cached-resolution-bug

Thanks for the repro project!
I tested on sbt 0.13.8-M6 and it seems to work now.
